Simplenote/ContentRecoveryManager.swift

```diff
@@ -0,0 +1,80 @@
+import Foundation
+import UniformTypeIdentifiers
+import CoreData
+
+public class ContentRecoveryManager {
+    private let fileManager: FileManager
+
+    public init(fileManager: FileManager = .default) {
+        self.fileManager = fileManager
+    }
+
+    private func createRecoveryDirIfNeeded() {
+        guard !fileManager.directoryExistsAtURL(fileManager.recoveryDirectoryURL) else {
+            return
+        }
+
+        try? fileManager.createDirectory(at: fileManager.recoveryDirectoryURL, withIntermediateDirectories: true)
```

Comment:
(Same note as in iOS), this could be a FileManager extension, and you'd avoid the need to call it in multiple spots. Sample!

```
extension FileManager {

    var entryUploadCacheURL: URL {
        let output = syncCachesURL.appendingPathComponent(FileManagerFolders.uploadCache, isDirectory: true)
        if directoryExistsAtURL(output) == false {
            try? createDirectory(at: output, withIntermediateDirectories: true)
        }

        return output
    }
```